wallet/addresses_test.go

```diff
@@ -0,0 +1,374 @@
+		},
+		{
+			address:     "SssBoVxTkCUb6xs7vph3BHdPmki3weVvRsF",
+			addressHash: hexToBytes("fa8073fcb670ba7312a1ef0d908cfb05c59b70b9"),
+			internal:    true,
+			compressed:  true,
+			pubKey:      hexToBytes("0327540e546f9cfac45f51699e2656732727507971060641ead554d78eeea88aa6"),
+		},
+	}
+
+	expectedExternalAddrs = []expectedAddr{
+		{
+			address:     "SsfPTmZmaXGkXfcNGjftRPmoGGCqtNPCHKx",
+			addressHash: hexToBytes("791376f67fb3becf392b071d25d7c99c82139ee3"),
+			internal:    false,
+			compressed:  true,
+			pubKey:      hexToBytes("031634efb3e83c834a82cdc898000f85215a09dc742d5b3b82ace7221ca1bb0938"),
+		},
+		{
+			address:     "SsXhSHBiaEan7Ls36bvhLspZ3LC1NKzuwQz",
+			addressHash: hexToBytes("24b8b3d89f987bf3cd80a8c16d9368d683217fa4"),
+			internal:    false,
+			compressed:  true,
+			pubKey:      hexToBytes("0280beb72c6ef42ce3133fd6d340fd5bedcfccaded5a6eabb6d2430e3958bf7c85"),
+		},
+		{
+			address:     "SspSfaWDNwc9TA31Q9iR2jot2eV1hk2ix6U",
+			addressHash: hexToBytes("dc67b3d95adb1789efe4aa73607d8a8c57eee2bb"),
+			internal:    false,
+			compressed:  true,
+			pubKey:      hexToBytes("03b120e0e073a12a1957680251a1562c5c6e30e547797fb5411107eac19699f601"),
+		},
+		{
+			address:     "SsrSYTB9MQQ1czAfPmWE66ZFqv7NrwzqfQT",
+			addressHash: hexToBytes("f252015c8e0059c21cae623704d8588d12ca5c74"),
+			internal:    false,
+			compressed:  true,
+			pubKey:      hexToBytes("0350822c9bd61f524f4d68fa605e850c34c5e8ccc9b5cf278782131c1e21dd261b"),
+		},
+		{
+			address:     "SsqBcGBre8SZrG61cd5M5e2GaMJbK2CMdEa",
+			addressHash: hexToBytes("e486d22d1244becac5a30b38dda1c8c4c1b3bdeb"),
+			internal:    false,
+			compressed:  true,
+			pubKey:      hexToBytes("031c494068c9c454bef7de35fa4717f21c07dec4471bd8500650b133d57e49a81d"),
+		},
+	}
+)
+
+func setupWallet(t *testing.T, cfg *Config) (w *Wallet, db walletdb.DB, teardown func()) {
+	f, err := ioutil.TempFile("", "testwallet.db")
+	if err != nil {
+		t.Fatal(err)
+	}
+	f.Close()
+	db, err = walletdb.Create("bdb", f.Name())
+	if err != nil {
+		t.Fatal(err)
+	}
+	rm := func() {
+		db.Close()
+		os.Remove(f.Name())
+	}
+	err = Create(opaqueDB{db}, pubPassphrase, privPassphrase, seed, cfg.Params)
+	if err != nil {
+		rm()
+		t.Fatal(err)
+	}
+	cfg.DB = opaqueDB{db}
+	w, err = Open(cfg)
+	if err != nil {
+		rm()
+		t.Fatal(err)
+	}
+	w.Start()
+
+	teardown = func() {
+		w.Stop()
+		w.WaitForShutdown()
+		rm()
+	}
+	return
+}
+
+func testExternalAddresses(tc *testContext) {
+	w, db, teardown := setupWallet(tc.t, &walletConfig)
+	defer teardown()
+
+	prefix := "testExternalAddresses"
+
+	if tc.watchingOnly {
+		err := walletdb.Update(db, func(tx walletdb.ReadWriteTx) error {
+			ns := tx.ReadWriteBucket(waddrmgrBucketKey)
+			return w.Manager.ConvertToWatchingOnly(ns)
+		})
+		if err != nil {
+			tc.t.Fatalf("%s: failed to convert wallet to watching only: %v",
+				prefix, err)
+		}
+	}
+
+	if tc.locked {
+		w.Lock()
```

Comment:
A just-opened wallet should already be locked, and the passphrase will need to be provided in order to unlock a non-locked test wallet.